src/Symfony/Component/Translation/PseudoLocalizationTranslator.php

```diff
@@ -0,0 +1,273 @@
+                'g' => 'ĝ',
+                'h' => 'ĥ',
+                'i' => 'î',
+                'j' => 'ĵ',
+                'k' => 'ķ',
+                'l' => 'ļ',
+                'm' => 'ɱ',
+                'n' => 'ñ',
+                'o' => 'ö',
+                'p' => 'þ',
+                'q' => 'ǫ',
+                'r' => 'ŕ',
+                's' => 'š',
+                't' => 'ţ',
+                'u' => 'û',
+                'v' => 'ṽ',
+                'w' => 'ŵ',
+                'x' => 'ẋ',
+                'y' => 'ý',
+                'z' => 'ž',
+                '{' => '(',
+                '|' => '¦',
+                '}' => ')',
+                '~' => '˞', ]) : $text;
+        }
+
+        if (1.0 < $this->expansionFactor) {
+            $textVisibleLength = false === ($encoding = mb_detect_encoding($visibleText, null, true)) ? \strlen($visibleText) : mb_strlen($visibleText, $encoding);
+            $transMissingLength = (int) (ceil($textVisibleLength * $this->expansionFactor)) - $textVisibleLength;
+            if ($this->brackets) {
+                $transMissingLength -= 2;
+            }
+
+            if (0 < $transMissingLength) {
+                $transPrefixLength = (int) (floor($transMissingLength / 2));
+                $transPrefix = '';
+                for ($i = 0; $i < $transPrefixLength; ++$i) {
+                    $transPrefix .= 0 === $i % 2 ? '~' : ' ';
+                }
+
+                $transSuffixLength = $transMissingLength - $transPrefixLength;
+                $transSuffix = '';
+                for ($i = 0; $i < $transSuffixLength; ++$i) {
+                    $transSuffix .= 0 === $i % 2 ? ' ' : '~';
+                }
+
+                $trans = $transPrefix.$trans.$transSuffix;
+            }
+        }
+
+        if ($this->brackets) {
+            $trans = '['.$trans.']';
+        }
+
+        return $trans;
+    }
+
+    private function getParts(string $originalTrans): array
+    {
+        if (!$this->parseHTML) {
+            return [[true, true, $originalTrans]];
+        }
+
+        $crawler = new Crawler();
+        $crawler->addHtmlContent('<html><body><trans>'.$originalTrans.'</trans></body></html>');
+
+        return $this->parseNode($crawler->children('body > trans')->getNode(0));
+    }
+
+    private function parseNode(\DOMNode $node): array
+    {
+        $parts = [];
+
+        foreach ($node->childNodes as $childNode) {
+            if (!$childNode instanceof \DOMElement) {
+                $parts[] = [true, true, $childNode->nodeValue];
+
+                continue;
+            }
+
+            $parts[] = [false, false, '<'.$childNode->tagName];
+
+            /** @var \DOMAttr $attribute */
+            foreach ($childNode->attributes as $attribute) {
+                $parts[] = [false, false, ' '.$attribute->nodeName.'="'];
+                $parts[] = [false, \in_array($attribute->nodeName, $this->localizableHTMLAttributes, true), $attribute->nodeValue];
```

Comment:
This is a wrong implementation. You take the parsed value, and you add it as a part of HTML. This means that you are removing any escaping here (as you don't apply it when rebuilding the HTML)

Reply:
You mean I need to take into account the fact that an attribute value can contain escaped chars such as quotes?

Reply:
Escaped htm special chars are now taken into account. They are not considered as localizable.